json.lua

```diff
@@ -108,7 +119,7 @@ local function encode_number(val)
   if val ~= val or val <= -math.huge or val >= math.huge then
     error("unexpected number value '" .. tostring(val) .. "'")
   end
-  return string.format("%.14g", val)
+  return tostring(val)
```

Comment:
Unreliable.

Reply:
Don't understand. Perhaps a more useful/productive comment? Thanks.

Reply:
Sorry, that was indeed too short to be useful. The documentation of `tostring` says:

> Receives an argument of any type and converts it to a string in a reasonable format. For complete control of how numbers are converted, use [string.format](https://www.lua.org/manual/5.1/manual.html#pdf-string.format).

It doesn't guarantee anything. It could output the number with arbitrarily poor precision, or it could use an arbitrary format (which need not be valid JSON). (In practice, it does indeed behave quite reasonable, but you shouldn't rely on this; it should also be noted that it is not "lossless" in practice, it rounds (albeit neither is the current method - 14 digits aren't enough).)

Reply:
Thank you for explaining it. I see. I remember I had changed it to make a certain case work properly when I was using it in the oauth2 module. Unfortunately I did not document why I changed it. I think I switched it back in the repository for now.